renderer/src/features/registry-servers/components/dynamic-array-field.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react'
+import { Button } from '@/common/components/ui/button'
+import { Input } from '@/common/components/ui/input'
+import { Label } from '@/common/components/ui/label'
+import { Trash2 } from 'lucide-react'
+
+interface DynamicArrayFieldProps {
+  label: string
+  value: string[]
+  onChange: (value: string[]) => void
+  inputLabelPrefix?: string
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>
+  addButtonText?: string
+  validate?: (value: string) => string | null
+}
+
+export const DynamicArrayField = ({
+  label,
+  value,
+  onChange,
+  inputLabelPrefix = 'Item',
+  inputProps = {},
+  addButtonText = 'Add',
+  validate,
+}: DynamicArrayFieldProps) => {
+  const [touched, setTouched] = useState<boolean[]>(value.map(() => false))
+
+  React.useEffect(() => {
+    // Adjusts "touched" array to the correct length
+    setTouched((t) => {
+      if (value.length > t.length)
+        return [...t, ...Array(value.length - t.length).fill(false)]
+      if (value.length < t.length) return t.slice(0, value.length)
+      return t
+    })
+  }, [value.length])
+
+  return (
+    <div className="mt-6 w-full">
+      <Label>{label}</Label>
+      <div role="group" aria-label={label} className="w-full">
+        <div className="mt-2 flex w-full flex-col gap-2">
+          {value.map((item, idx) => {
+            const error = validate ? validate(item) : null
+            return (
+              <div key={idx} className="flex w-full flex-col gap-1">
+                <div className="flex w-full items-center gap-2">
+                  <Input
+                    type="text"
```

Comment:
I guess it is better duplicating the fields rather  than use `DynamicArrayField` with config, cause using react-hook-form is really easy to do it, and you can use soz for validation.
Anyways in case you don't want to use react-hook-form I would dynamic set the type 
`type="number"` based on fields, in this way you don't have
<img width="568" height="115" alt="Screenshot 2025-07-14 at 15 55 22" src="https://github.com/user-attachments/assets/bfdf40d9-daef-48de-8377-7c6d38dae9eb" />